### PR TITLE
[feat] Support web3 websocket provider and passing web3 in via config

### PR DESCRIPTION
**Pull request by dylanseago**
- Allowing web3 to be passed into config fixes the memory leak issue we've been seeing. Each instance of Web3 appears to leak around 112kB of non-heap memory. Treating it as a singleton by passing in an existing web3 instance provides a solution
- Adds support for websocket providers
- Upgrades to most recent dependencies for many libraries

### CODE REVIEW

**Comment by dzdidi** [comment]
I do not understand why do we need `web3` and `server` as properties of this class?

How (serialised?) web3 will look like as a part of config?

File: packages/ethereum-payments/src/BaseEthereumPayments.ts [hunk 1]
```diff
@@ -60,13 +61,23 @@ implements BasePayments
...
-
     this.config = config
-    this.eth = web3.eth
+    this.server = config.fullNode || null
+    if (config.web3) {
+      this.web3 = config.web3
+    } else if (isNull(this.server)) {
+      this.web3 = new Web3()
+    } else if (this.server.startsWith('http')) {
+      this.web3 = new Web3(new Web3.providers.HttpProvider(this.server, config.providerOptions))
```
Comment (dzdidi) [inline packages/ethereum-payments/src/BaseEthereumPayments.ts:71]: Wasn't `transactionConfirmationBlocks` needed for ganache tests to pass?

**Comment by dzdidi** [comment]
Also it seems like there is something off with dependencies:
```
> cd packages/ethereum-payments
> npm install
> cd ../..
> npm install
> node_modules/lerna/cli.js run --scope @faast/ethereum-payments test --bail                                                                                                                                                                                                                               [12:37:11]
lerna notice cli v3.18.3
lerna notice filter including "@faast/ethereum-payments"
lerna info filter [ '@faast/ethereum-payments' ]
lerna info Executing command in 1 package: "npm run test"
lerna ERR! npm run test exited 1 in '@faast/ethereum-payments'
lerna ERR! npm run test stdout:

> @faast/ethereum-payments@2.7.0 test /Users/denysbitaccess/Projects/coin-payments/packages/ethereum-payments
> jest --runInBand --detectOpenHandles --bail


lerna ERR! npm run test stderr:
TypeError: _gracefulFs(...).realpathSync.native is not a function
    at tryRealpath (/Users/denysbitaccess/Projects/coin-payments/packages/ethereum-payments/node_modules/jest-util/build/tryRealpath.js:26:39)
    at getCacheDirectory (/Users/denysbitaccess/Projects/coin-payments/packages/ethereum-payments/node_modules/jest-config/build/getCacheDirectory.js:89:33)
    at Object.<anonymous> (/Users/denysbitaccess/Projects/coin-payments/packages/ethereum-payments/node_modules/jest-config/build/Defaults.js:39:50)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @faast/ethereum-payments@2.7.0 test: `jest --runInBand --detectOpenHandles --bail`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @faast/ethereum-payments@2.7.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/denysbitaccess/.npm/_logs/2020-07-23T10_37_52_689Z-debug.log

lerna ERR! npm run test exited 1 in '@faast/ethereum-payments'
```

File: packages/ethereum-payments/src/BaseEthereumPayments.ts (hunk 1 above)
Comment (dylanseago) [reply packages/ethereum-payments/src/BaseEthereumPayments.ts:71]: Not needed

**Comment by dylanseago** [comment]
Both web3 and server need to be exposed to support server side caching of the web3 instance. Every other payments lib does the same with similar names. We could combine the web3 and fullNode config options into one but there's no easy way I can see to retrieve the fullNode url from a web3 instance.

Web3 is excluded from the public config through omit call so it won't have serialization issues

**Pull request merged**